lib/utils.ex

```diff
@@ -0,0 +1,24 @@
+defmodule HashRing.Utils do
+  @compile :native
+
+  @spec hash(atom | binary | integer) :: integer
+  def hash(key) when is_binary(key) do
+    <<_ :: binary-size(8), value :: unsigned-little-integer-size(64)>> = :erlang.md5(key)
+    value
+  end
+  def hash(key), do: hash("#{key}")
+
+  @spec gen_items(binary, integer) :: [{integer, binary}]
+  def gen_items([], _num_replicas), do: {}
+  def gen_items(nodes, num_replicas) do
+    gen_items(nodes, Enum.to_list(0..(num_replicas - 1)), [])
+  end
+
+  defp gen_items([], _replicas, items) do
+    Enum.sort(items, &(elem(&1, 0) < elem(&2, 0)))
```

Comment:
To re-open this, 
Are you concerned about the speed? Because you shouldn't be. 
I benchmarked this using benchfella. Sorting a million element shuffled list of a tuple of ten random integers, you get this:
```elixir
## SortingBench
benchmark name             iterations   average time
Enum.sort(elems)                    1   1555209.00 µs/op
Enum.sort(elems, sort_fn)           1   2808098.00 µs/op
```
The custom sorter is significantly slower. It gets worse if the first element is a string or an atom. 
And it's harder to read.

You only gain benefit if every single tuple has the same element in the first position, if there are even two choices, (basically modding the first element by 2) the default sorter is faster.

Reply:
I'm not concerned about the speed. Just didn't think this was worth touching in the scope of this PR.